irohad/ametsuchi/block_query_factory.hpp

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_BLOCK_QUERY_FACTORY_HPP
+#define IROHA_BLOCK_QUERY_FACTORY_HPP
+
+#include <boost/optional.hpp>
+
+#include "ametsuchi/block_query.hpp"
+
+namespace iroha {
+  namespace ametsuchi {
+    class BlockQueryFactory {
+     public:
+      /**
+       * Creates a block query from the current state.
+       * @return Created block query
+       */
+      virtual boost::optional<std::shared_ptr<BlockQuery>>
+      createBlockQuery() = 0;
```

Comment:
Can't this method be `const`?

Reply:
This method is overloaded in storage and in mocks

Reply:
It has some issues with mocks than